packages/toggleInputList/tests/ToggleInputList.test.tsx

```diff
@@ -15,78 +16,87 @@ expect.addSnapshotSerializer(createSerializer());
 
 describe("ToggleInputList", () => {
   it("renders default", () => {
-    const component = mount(
+    const { getByText, getAllByRole, asFragment } = render(
       <ToggleInputList
         id="checkbox"
         items={options}
         listLabel="Sample legend"
       />
     );
-    expect(toJson(component)).toMatchSnapshot();
+    getByText("Sample legend");
+    expect(getAllByRole("checkbox").length).toBe(3);
+    expect(asFragment()).toMatchSnapshot();
   });
 
   it("renders as a radio group", () => {
-    const component = mount(
+    const { asFragment } = render(
       <ToggleInputList
         id="radio"
         items={options}
         listLabel="Sample legend"
         isRadioGroup={true}
       />
     );
-    expect(toJson(component)).toMatchSnapshot();
+    expect(asFragment()).toMatchSnapshot();
   });
 
   it("renders with selected items", () => {
-    const component = mount(
+    const { getAllByRole } = render(
       <ToggleInputList
         id="checkbox"
         items={options}
         listLabel="Sample legend"
         selectedItems={["value.1", "value.2"]}
       />
     );
-    expect(toJson(component)).toMatchSnapshot();
+    const checkboxes = getAllByRole("checkbox") as HTMLInputElement[];
+    const option1 = checkboxes[0];
+    const option2 = checkboxes[1];
+    const option3 = checkboxes[2];
+    expect(option1.checked).toBeTruthy();
+    expect(option2.checked).toBeTruthy();
+    expect(option3.checked).toBeFalsy();
   });
 
   it("renders with errors", () => {
-    const component = mount(
+    const { getByText } = render(
       <ToggleInputList
         id="checkbox"
         items={options}
         listLabel="Sample legend"
         errors={["error.1", "error.2"]}
       />
     );
-    expect(toJson(component)).toMatchSnapshot();
+    getByText("error.1");
+    getByText("error.2");
   });
 
   it("renders with hidden label", () => {
-    const component = mount(
+    const { queryByText } = render(
       <ToggleInputList
         id="checkbox"
         items={options}
         listLabel="Sample legend"
         showListLabel={false}
       />
     );
-    expect(toJson(component)).toMatchSnapshot();
+    expect(queryByText("Sample legend")).not.toBeVisible();
   });
 
   it("renders with required", () => {
-    const component = mount(
+    const { getByTestId } = render(
       <ToggleInputList
         id="checkbox"
         items={options}
         listLabel="Sample legend"
         required={true}
       />
     );
-    expect(toJson(component)).toMatchSnapshot();
+    getByTestId("dangerText");
```

Comment:
Should there be an assertion of some sort here?

Reply:
It fails if it does not find the element.

Reply:
Couldn't this also check for the required prop also given - `"renders with required"`? The danger test existing does confirm but I wonder if it could be more specific.

Reply:
@nataliepina didn't get you. With react testing library, we intend to test what the user perceives. So, I don't think we should test any prop. So, ideally here we should check for asteric sign for the required field. But it was getting difficult to find svg. So, instead of finding SVG tested with it's test-id. Do let me know if there is any action item on this thread.